python/lsst/pipe/base/argumentParser.py

```diff
@@ -732,19 +724,28 @@ def __call__(self, parser, namespace, values, option_string):
         @param[in] parser       argument parser (instance of ArgumentParser)
         @param[in] namespace    parsed command (an instance of argparse.Namespace); ignored
         @param[in] values       a list of trace levels;
-            each item must be of the form component_name=level
+            each item must be of the form component_name=level or level
+            where level is an integer or a keyword
         @param[in] option_string    option value specified by the user (a str)
         """
+        permittedLevelList = ('DEBUG', 'INFO', 'WARN', 'FATAL')
```

Comment:
Casting a set to a list loses its order, and the order is useful in the error message. I agree it is annoying to save two such similar variable, but I didn't think of a simple, fast alternative. Two other solutiosn I did think of:
- Cast to a set for each test, or search the list instead of the set. Neither of these slows parsing very much, but it irks me to use linear search.
-  Save the info as a dict of name: severity, then sort by severity when printing the message. In my opinion this adds too much code.